src/cspmonitor/CSPMonitor.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+
+export default class CSPMonitor {
+  private static loggers: Set<Logger> = new Set<Logger>();
+  private static shouldRegisterCSPMonitor: boolean = true;
+  private static added = false;
+
+  /* istanbul ignore next */
+  static register(): void {
+    if (!('document' in global) || !document.addEventListener) {
+      return;
+    }
+    if (CSPMonitor.shouldRegisterCSPMonitor) {
+      if (!CSPMonitor.added) {
+        document.addEventListener('securitypolicyviolation', CSPMonitor.listener);
+        CSPMonitor.added = true;
+      }
+    }
+  }
+
+  /* istanbul ignore next */
+  static unregister(): void {
+    if (!('document' in global) || !document.removeEventListener) {
+      return;
+    }
+    document.removeEventListener('securitypolicyviolation', CSPMonitor.listener);
+    CSPMonitor.loggers = new Set<Logger>();
+    CSPMonitor.added = true;
+  }
+
+  static disable(): void {
+    CSPMonitor.shouldRegisterCSPMonitor = false;
+    CSPMonitor.unregister();
+  }
+
+  static addLogger(logger: Logger): void {
+    CSPMonitor.loggers.add(logger);
+  }
```

Comment:
We run without strict mode in TypeScript, so please add the check here:

```
if (logger) {
  …
}
```

Reply:
Will add.